apps/emqx_resource/src/emqx_resource_manager.erl

```diff
@@ -1706,6 +1804,19 @@ handle_retry_add_channel(Data0, ChannelId) ->
             keep_state_and_data
     end.
 
+handle_retry_remove_channel(Data0, ChannelId) ->
+    ?tp(retry_remove_channel, #{channel_id => ChannelId}),
+    From = undefined,
+    maybe
+        {ok, _} ?= maps:find(ChannelId, Data0#data.added_channels),
```

Comment:
Nit:
m.b. for more transparent consistency and introspection, we could keep all the pending events and timers in the `added_channels` map, together with channel status.

Reply:
it's possible to introspect the timers with `sys:get_status/1`, which might be better, since we also may have postponed events which will not show up with `sys:get_state/1`, if that's the intention